drivers/wireless/bluetooth/bt_rpmsghci.c

```diff
@@ -0,0 +1,535 @@
+{
+  FAR struct rpmsghci_s      *priv = NULL;
+  FAR struct rpmsghci_open_s *msg  = NULL;
+  uint32_t                    space = 0;
+  int                         ret  = OK;
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->rpmsg;
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      ret = -ENOMEM;
+      goto errout;
+    }
+
+  ret = rpmsghci_send(priv, RPMSGHCI_OPEN, &msg->header,
+                      sizeof(struct rpmsghci_open_s));
+errout:
+  return ret;
+}
+
+/****************************************************************************
+ * Name: bt_rpmsghci_close
+ *
+ * Description:
+ *   Rpmsg-HCI close operation
+ *
+ * Parameters:
+ *   btdev  - the bt instance
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static void bt_rpmsghci_close(struct bt_driver_s *btdev)
+{
+  FAR struct rpmsghci_s       *priv  = NULL;
+  FAR struct rpmsghci_close_s *msg   = NULL;
+  uint32_t                     space = 0;
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->rpmsg;
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      return;
+    }
+
+  rpmsghci_send(priv, RPMSGHCI_CLOSE, &msg->header,
+                sizeof(struct rpmsghci_close_s));
+}
+
+/****************************************************************************
+ * Name: bt_rpmsghci_send
+ *
+ * Description:
+ *   Rpmsg-HCI send operation
+ *
+ * Parameters:
+ *   btdev  - the bt instance
+ *   type   - bt command type
+ *   data   - bt command data
+ *   len    - bt command data length
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int bt_rpmsghci_send(struct bt_driver_s *btdev,
+                            enum bt_buf_type_e type,
+                            void *data, size_t len)
+{
+  FAR struct rpmsghci_s      *priv  = NULL;
+  FAR struct rpmsghci_data_s *msg   = NULL;
+  uint32_t                    space = 0;
+  int                         ret   = OK;
+
+  wlinfo("bt_rpmsghci_send %d\n", type);
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->rpmsg;
+
+  if (data == NULL)
+    {
+      ret = -EINVAL;
+      goto errout;
+    }
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      ret = -ENOMEM;
+      goto errout;
+    }
+
+  DEBUGASSERT(sizeof(struct rpmsghci_data_s) - 1 + len <= space);
+  memcpy(msg->data, data, len);
+  msg->type = type;
+  ret = rpmsghci_send(priv, RPMSGHCI_SEND, &msg->header,
+                      sizeof(struct rpmsghci_data_s) -1 + len);
```

Comment:
let's skip the response to improve the speed?

Reply:
This one implements `send()` operation for `struct bt_driver_s`, so it's better to now if the sending operation was success or not. BTW, in the case of HCI, the speed of operations is probably not so important.